neo/Cryptography/ECC/ECFieldElement.cs

```diff
@@ -26,28 +26,24 @@ public int CompareTo(ECFieldElement other)
 
         public override bool Equals(object obj)
         {
-            if (obj == this)
-                return true;
-
-            ECFieldElement other = obj as ECFieldElement;
-
-            if (other == null)
-                return false;
-
+            if (!(obj is ECFieldElement other)) return false;
+            if (ReferenceEquals(this, obj)) return true;
             return Equals(other);
         }
 
         public bool Equals(ECFieldElement other)
         {
-            return Value.Equals(other.Value);
+            return Value.Equals(other.Value) && curve.Equals(other.curve);
```

Comment:
Don't need to check this curve here Shargon... comparing points and curves it's different things. I mean, we can have the same point (equals) on different curves each... it's unnecessary overhead.

Reply:
For me have no sense two different objects with different values and return true in equal.

Reply:
I'm with Shargon. For the "Equals" implementation I think we should compare all properties. 
We can have a different method if we want to optimize this check